confgenerator/confmerger.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package confgenerator provides functions to generate subagents configuration from unified agent.
+package confgenerator
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+var (
+	builtInConf = map[string]string{
```

Comment:
Our config is all structured data. Why is this a string instead of just a Config struct directly? If you really want to write it as a file you can always just dump the yaml (or even better, you can dump the fully merged Config struct).

Reply:
Yup, this is something we can definitely improve. Dumping yaml makes it easier for users to debug, right? Because their config is yaml based, so it's easier to diagnose the `built-in yaml` + `user specififed yaml` => `merged yaml` flow.

Reply:
I think if users need to look at the generated config to debug what's happening then something has already gone horribly wrong with the design. It should be abundantly obvious to the user from looking only at the config file they control what the Ops Agent is going to do.

Reply:
Hopefully per offline discussion, it's clear that this is kinda inevitable given the features we'd like to provide.